services/tables/src/main/java/com/linkedin/openhouse/tables/utils/TableUUIDGenerator.java

```diff
@@ -143,11 +143,13 @@ private void validatePathOfProvidedRequest(
     String tblIdFromProps =
         extractFromTblPropsIfExists(databaseId + "." + tableId, tableProperties, TBL_RAW_KEY);
 
-    java.nio.file.Path previousPath =
-        InternalRepositoryUtils.constructTablePath(
-            storageManager, dbIdFromProps, tblIdFromProps, tableUUIDProperty);
-    if (TableType.REPLICA_TABLE != tableType && !doesPathExist(previousPath)) {
-      log.error("Previous tableLocation: {} doesn't exist", previousPath);
+    Storage storage = catalog.resolveStorage(TableIdentifier.of(dbIdFromProps, tblIdFromProps));
```

Comment:
resolveStorage will use `SelectStorage` if the entry doesn't exist in HTS. That code is not guaranteed to be idempotent during migration, lets avoid using selectStorage again in the consecutive COMMIT/PutSnapshot call.

Can we instead use tableLocation and derive storage from there? ie:
if s3://a/b/c -> S3Storage
if abs://a/b/c -> ABSStorage
if a/b/c -> HDFSStorage

Reply:
done. Please check. there is a catch though, because both local and hdfs storage have same paths